core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -92,6 +92,57 @@ class MultiCloudWorkspaceService(override val ctx: RawlsRequestContext,
     with Retry
     with WorkspaceSupport {
 
+  /**
+    * Deletes a workspace. For legacy "rawls" workspaces,
+    * delegates to the WorkspaceService codepath. For MC workspaces (i.e., azure),
+    * initiates a deletion with WorkspaceManager and then deletes the underlying rawls record.
+    * @param workspaceName Tuple of workspace name and namespace
+    * @param workspaceService WorkspaceService instance which will handle deletion of GCP workspaces
+    * @return If GCP, name of the GCP bucket being deleted; otherwise None
+    */
+  def deleteMultiCloudOrRawlsWorkspace(workspaceName: WorkspaceName,
+                                       workspaceService: WorkspaceService
+  ): Future[Option[String]] =
+    for {
+      workspace <- getWorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.delete)
+      _ = logger.info(
+        s"Deleting workspace [workspaceId=${workspace.workspaceId}, name=${workspaceName.name}, billingProject=${workspace.namespace}, user=${ctx.userInfo.userSubjectId.value}]"
+      )
+      result: Option[String] <- workspace.workspaceType match {
+        case RawlsWorkspace => workspaceService.deleteWorkspace(workspaceName)
+        case McWorkspace    => deleteMultiCloudWorkspace(workspace)
+      }
+    } yield result
+
+  private def deleteMultiCloudWorkspace(workspace: Workspace): Future[Option[String]] =
+    for {
+      _ <- deleteWorkspaceInWSM(workspace.workspaceIdAsUUID).recover { case e: ApiException =>
+        if (e.getCode == StatusCodes.NotFound.intValue) {
```

Comment:
There isn't any plausibility of a user being able to delete a workspace in one service but not the other, right? 
Just mentioning it because a lot of the time, services will return 404 if the user can't access the resource. 
I don't think there's actually a problem here: just bringing it up in case I'm wrong about something.

Reply:
404 in WSM generally means the resource is not there. WSM will return a 403 otherwise.

Reply:
Also, the `workspace` resource we're checking the `delete` action against is the same resource WSM will check against and delete. So if Sam tells Rawls that the user can delete the workspace, Sam will tell WSM the same thing